facebook/OpenBIC
# fby35: hda1: Support mctp-pldm for mpro

Summary:
- Enable mctp-pldm channel for Ampere Mpro device.
- Support 4byte post code oem command (NetFn:0x38, Cmd:0x2A)
- Support mctp-pldm shell command in BIC console.
- Support platform event message command for hda1.
- Support pldm request bridge(pldm->ipmb) command for Mpro to BMC.
- Support pldm request bridge(ipmb->pldm) command for BMC to Mpro.

Test Plan:
- BuildCode: PASS
- Check PLDM request command from BIC to Mpro: PASS
- Check PLDM request command from BMC to Mpro: PASS
- Check get post code command from BMC: PASS

Log:
- Check PLDM request command from BIC to Mpro:
  ```
  uart:~$ platform pldm sendreq 0x10 0x02 0x11 0x03 0x00 0x00
  * mctp: 0x44ff8 addr: 0x9e eid: 0x10 msg_type: 0x1
    pldm_type: 0x2 pldm cmd: 0x11
  00000000: 03 00 00                                         |...              |
  00000000: 00 05 00 02 01 00 01 2f  00 00 00                |......./ ...     |
  ``` 

- Check PLDM request command from BMC to Mpro
  ```
  root@bmc-oob:~# bic-util slot1 0xe0 0x02 0x15 0xa0 0x00 0x42 0x02 0x11 0x03 0x00 0x00
  15 A0 00 00 05 00 02 01 00 01 2f 00 00 00
  ```

- Check get post code command from BMC:
  ```
  root@bmc-oob:~# bic-util slot1 --get_gpio |grep -i POST
  11 AMP_FM_BIOS_POST_CMPLT_BIC_N: 0
  root@bmc-oob:~# bic-util slot1 --get_post_code
  util_get_post_buf: returns 38 dword
  [03101019] [03051001] [03051002] [03051001] 
  [03051002] [03051001] [03051002] [03051001] 
  [03051002] [03051001] [02020000] [02010000] 
  [02020000] [02020003] [02020004] [02020000] 
  [02010000] [02011001] [02010000] [03051005] 
  [03041001] [03021001] [97808001] [98808001] 
  [98808000] [97808000] [93808001] [93808000] 
  [94808001] [95808002] [95804201] [95802101] 
  [95808000] [94808000] [92808001] [92808000] 
  [91808001] [91808000] 
  ```
